lib/beaker-answers/pe_conf.rb

```diff
@@ -157,10 +182,40 @@ def the_host_with_role(role)
         Array(h['roles']).include?(role.to_s)
       end
 
-      if found_hosts.length == 0 or found_hosts.length > 1
+      if (found_hosts.length == 0 or found_hosts.length > 1) and (role != 'pe_postgres')
```

Comment:
If there is no host with pe_postgres then an error is raised. 
Since pe_postgres is optional, we shouldn't raise that error.
Hurm, yes I see your point, if we have more then one pe_postgres we should raise a specific error. I'll work on that and change out the &&

Reply:
I see, so we have two ways of calling the_host_with_role(), in one case we consider it exceptional if we can't find a node, and in the other we don't.  Rather than specialize around this method understanding about pe_postgres, let's pass it an option to raise_exceptions.  For more idiomatic ruby, could default that to false, and add a the_host_with_role!(role) which calls the_host_with_role(role, true), and sub the_host_with_role!() wherever we are currently using it (except the times we are checking for pe_postgres).